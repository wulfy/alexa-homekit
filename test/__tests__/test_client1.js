const base_config = require("../config_tests/base_config");

const { DOMOTICZ_GET_DEVICES } = require("../mockups/client1Mockup");
global.getDomoticzFromToken = (token) => {
    return new base_config.mockedDomoticz(token,DOMOTICZ_GET_DEVICES);
}

//TEST FOR INDEX.JS
test('DISCOVERY TESTING client1', done => {
    let context = {};
    context.succeed = function (data){
        const testData = JSON.stringify(data); 
        expect(testData).toMatchSnapshot();
        done();
    };
    
    base_config.handler(base_config.mockups.ALEXA_DISCOVERY_REQUEST_EXAMPLE,context);
});