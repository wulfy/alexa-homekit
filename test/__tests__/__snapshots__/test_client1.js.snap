// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DISCOVERY TESTING client1 1`] = `"{\\"event\\":{\\"header\\":{\\"namespace\\":\\"Alexa.Discovery\\",\\"name\\":\\"Discover.Response\\",\\"payloadVersion\\":\\"3\\",\\"messageId\\":\\"abc-123-def-456\\"},\\"payload\\":{\\"endpoints\\":[{\\"endpointId\\":\\"155_tempHubharmony_OnOff\\",\\"manufacturerName\\":\\"temp / Hub  + harmony\\",\\"friendlyName\\":\\"robot mi naomi 27 \\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"TEMPERATURE_SENSOR\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.TemperatureSensor\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"temperature\\",\\"value\\":\\"()=>({value:undefined,scale:'CELSIUS'})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":false}}]},{\\"endpointId\\":\\"146_zwave_BlindsPercentage\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"volet salon\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Open' === 'Off' || 'Open' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"18_zwave_BlindsPercentage\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"volet Gbaie\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Open' === 'Off' || 'Open' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"151_zwave_BlindsPercentage\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"volet Pbaie\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Open' === 'Off' || 'Open' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"159_zwave_BlindsPercentage\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"volet couloir\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Open' === 'Off' || 'Open' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"164_zwave_BlindsPercentage\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"volet cellier\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Open' === 'Off' || 'Open' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"170_mirobot_Selector\\",\\"manufacturerName\\":\\"mi.robot\\",\\"friendlyName\\":\\"mi robot   Control\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{\\"MaxDimLevel\\":\\"100\\",\\"ReversePosition\\":\\"undefined\\"},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Off' === 'Off' || 'Off' === 'Closed') ? ('undefined' === 'true' ? 'ON' : 'OFF') : ('undefined' === 'true' ? 'OFF' : 'ON')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"206_zwave_undefined\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"TemperatureSejour\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"TEMPERATURE_SENSOR\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.TemperatureSensor\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"temperature\\",\\"value\\":\\"()=>({value:20.1,scale:'CELSIUS'})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":false}}]},{\\"endpointId\\":\\"200_zwave_undefined\\",\\"manufacturerName\\":\\"zwave\\",\\"friendlyName\\":\\"TemperatureNuit\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"TEMPERATURE_SENSOR\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.TemperatureSensor\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"temperature\\",\\"value\\":\\"()=>({value:18.1,scale:'CELSIUS'})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":false}}]},{\\"endpointId\\":\\"209_dummydev_undefined\\",\\"manufacturerName\\":\\"dummy dev\\",\\"friendlyName\\":\\"Temperature Piscine\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"TEMPERATURE_SENSOR\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.TemperatureSensor\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"temperature\\",\\"value\\":\\"()=>({value:19.6,scale:'CELSIUS'})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":false}}]},{\\"endpointId\\":\\"256_RFplayer_undefined\\",\\"manufacturerName\\":\\"RF player\\",\\"friendlyName\\":\\"temperature exterieur\\",\\"description\\":\\"(pas de description)\\",\\"displayCategories\\":[\\"TEMPERATURE_SENSOR\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.TemperatureSensor\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"temperature\\",\\"value\\":\\"()=>({value:7.5,scale:'CELSIUS'})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":false}}]}]}}}"`;
